DifferentialDrive.md

```diff
@@ -0,0 +1,545 @@
+#### Resetting the Encoders
+
+At the start of the match (or any time you need to reset the robot's position), it's important to reset the encoder values to zero. This ensures that your distance calculations start from a known point.
+
+We'll reset the encoders during the subsystem initialization:
+```java 
+leftEncoder.setPosition(0);
+rightEncoder.setPosition(0);
+```
+#### adding a gyroscope 
+`AnalogGyro` requires a channel port which we store it in the ports file. 
+
+Let’s add the AnalogGyro to the existing drive.java file:
+```java 
+private final AnalogGyro gyro = new AnalogGyro(gyroChannel);
+```
+#### Resetting the Gyro
+
+At the start of the match (or anytime you need to ensure accurate heading data), it's important to reset the gyro. This removes any bias or drift in the sensor, ensuring that your heading starts from a value of 0.
+```java
+gyro.reset();
+```
+#### Introducing Odometry
+Position Tracking: Odometry continuously updates the robot's position, which is crucial for autonomous navigation.
+
+Orientation Awareness: It tracks the robot’s heading, allowing for more precise maneuvers.
+```java
+private final DifferentialDriveOdometry odometry;
+```
+#### Initializing Odometry
+
+Odometry needs an initial orientation for the robot. We can use the `getRotation2d()` method from our gyro for the current orientation.
+
+Here’s how we initialize the odometry in the constructor:
+```java
+odometry = new DifferentialDriveOdometry(
+        gyro.getRotation2d(), 
+        leftEncoder.getPosition(), 
+        rightEncoder.getPosition(), 
+        new Pose2d(new Translation2d(), new Rotation2d()));
+```
+`new Rotation2d()` and `new Translation2d()` simply returns a value of 0. Which both give the starting position of the robot. This can be configured based on how and where you're robot will be starting each match.
+#### Periodically Updating Odometry
+
+To keep track of the robot’s position in real-time, we need to update the odometry regularly with the latest encoder readings. This will typically happen in the periodic() method, ensuring that position data is always up-to-date.
+
+Here's how you might update odometry :
+```java
+public void updateOdometry(){
+    odometry.update(
+        gyro.getRotation2d(),             
+        leftEncoder.getPosition(),        
+        rightEncoder.getPosition()        
+    );
+}
+```
+Aside from updating our odometry, on some cases we might need to reset it. I will leave this up to you to figure out how to do. (Hint: `odometry` has a method called resetOdometry). We won't need it for this project but if you're using the drive for autos example, it is good to reset your odometry whenever autos start. 
+#### Using `periodic()`
+To ensure that our odometry is constantly updating as the robot moves, we run the update odometry method periodically. Which is basically every tick (0.02 seconds). This will go in the `periodic()`method that our drive class obtains from inherting the `subsystemBase` class.
```

Comment:
```suggestion
To ensure that our odometry is constantly updating as the robot moves, we run the `updateOdometry()` method periodically every tick (0.02 seconds by default). This will go in the `periodic()` method that all subsystems inherit from SubsystemBase, including our drive class.
```